plugins/uefi-capsule/fu-uefi-common.c

```diff
@@ -179,9 +179,17 @@ fu_uefi_get_esp_path_for_os(void)
 	const gchar *os_release_id = NULL;
 	const gchar *id_like = NULL;
 	g_autofree gchar *esp_path = NULL;
+	g_autofree gchar *systemd_path = NULL;
 	g_autoptr(GError) error_local = NULL;
-	g_autoptr(GHashTable) os_release = fwupd_get_os_release(&error_local);
+	g_autoptr(GHashTable) os_release = NULL;
+
+	/* distro (or user) is using systemd-boot */
+	systemd_path = g_build_filename("EFI", "systemd", NULL);
+	if (g_file_test(systemd_path, G_FILE_TEST_IS_DIR))
```

Comment:
don't you need to add the ESP location if g_file_test() is going to work? I think the systemd check needs to be in the caller for fu_uefi_get_esp_path_for_os, not in fu_uefi_get_esp_path_for_os itself.

Reply:
Oh I see what you mean.  You're right it would need to be the caller, and that does become more complicated.  Let me have another thought on it if it's really worth the effort.